sdk/core/core-client-rest/test/multipart.spec.ts

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { describe, it, assert } from "vitest";
+import { PartDescriptor, buildBodyPart } from "../src/multipart.js";
+import { stringToUint8Array } from "../../core-util/dist/commonjs/bytesEncoding.js";
+
+describe("multipart buildBodyPart", () => {
+  describe("content-type calculation", () => {
+    it.each([
+      {
+        description: "content type header set to null results in header being unset",
+        descriptor: {
+          contentType: null,
+          body: "even though the body is a string, the content type should still not be set",
+        },
+        expected: undefined,
+      },
+      {
+        description: "content-type header set in part headers takes precedence over contentType",
+        descriptor: {
+          contentType: "dont-pick-me",
+          headers: {
+            "content-type": "pick-me-instead",
+          },
+        },
+        expected: "pick-me-instead",
+      },
+      {
+        description: "contentType value takes precedence over natural content type",
+        descriptor: {
+          contentType: "application/json",
+          body: `{ "aaa": "bbb" }`, // would otherwise expect this to give text/plain; charset=UTF-8
+        },
+        expected: "application/json",
+      },
+      {
+        description: "content type is taken from Blob if available",
+        descriptor: {
+          body: new Blob([], { type: "content-type-from-blob" }),
+        },
+        expected: "content-type-from-blob",
+      },
+      {
+        description: "contentType value takes precedence over Blob content type",
+        descriptor: {
+          contentType: "pick-me",
+          body: new Blob([], { type: "content-type-from-blob" }),
+        },
+        expected: "pick-me",
+      },
+      {
+        description: "content type for Blob defaults to application/octet-stream",
+        descriptor: {
+          body: new Blob([]),
+        },
+        expected: "application/octet-stream",
+      },
+      {
+        description: `content type for Uint8Array defaults to application/octet-stream`,
```

Comment:
a case for headers?

Reply:
Do you have a specific case in mind?

Reply:
oh i mean the values in headers would override the content type in other places.

Reply:
Yeah, we have that covered [here](https://github.com/Azure/azure-sdk-for-js/pull/29047/files#diff-58e1d91ef54ca717c5a1c7c9dc6a0097d691e796e6eba9d4c8edef25caa051b7R19-R28) already